src/map.cpp

```diff
@@ -863,9 +863,9 @@ int_fast32_t AStarNodes::getTileWalkCost(const Creature& creature, const Tile* t
 // Floor
 Floor::~Floor()
 {
-	for (uint32_t i = 0; i < FLOOR_SIZE; ++i) {
-		for (uint32_t j = 0; j < FLOOR_SIZE; ++j) {
-			delete tiles[i][j];
+	for (auto& tile : tiles) {
+		for (auto& j : tile) {
```

Comment:
Well, assuming the two-dimensional array is contiguous, I find it nicer to be:
```
for (Tile* ptr = tiles[0][0]; ptr < tiles[FLOOR_SIZE][FLOOR_SIZE] + 1; ++ptr) {
  delete ptr;
}
```
Since it has only 1 level.